### DISCUSSION TITLE
Inquiry about Elastix's feature detection and registration

### DISCUSSION THREAD

**Discussion by msshojaei90** [SuperElastix/elastix#910] [General]
Hi everyone,

I have used Elastix for image registration, and the results are very good at matching features between two modalities compared to feature-based registration algorithms available online. From a technical perspective, I am curious to know how it can successfully match features in paired modalities, considering it is an intensity-based registration and not a feature-based registration.

I would greatly appreciate your help in understanding this point.

Best wishes,
Javad

**Comment by ntatsisk** [comment]
Hi Javad, great to hear that you found a good solution for your application using elastix!

As you correctly mentioned elastix is intensity-based and, hence, performs the registration by optimizing a similarity metric computed using pixel intensities from the image pair. If you consider [a simple registration example of a 2D rectangle](https://github.com/InsightSoftwareConsortium/ITKElastix/blob/main/examples/ITK_Example21_ConstrainedRegistration.ipynb), you can imagine that the optimal displacement/transform required to match the two rectangles so that the similarity metric is optimized, is also the position where the edges (aka "features") of the rectangles also match. So, in a sense the feature matching happens implicitly in elastix and not directly as in feature-based algorithms.

You can read more on the metric optimization in the elastix manual: https://elastix.lumc.nl/download/elastix-5.1.0-manual.pdf
